### ISSUE TITLE
Datepickers and Dropdowns cannot be set to readonly

### ISSUE THREAD

**Issue by pallisj** [infor-design/enterprise-ng#1466]
Datepickers and dropdown fields that are set to disabled/readonly in HTML are not being reflected as readonly fields.

**Expected behavior**
If a datepicker or dropdown field are set to disabled/readonly in the HTML then the fields should show up as uneditable.

**Version**
- ids-enterprise-ng: 15.2.1
- ids-enterprise-typings 15.2.1
- ids-enterprise 4.82.0

![image](https://user-images.githubusercontent.com/129091848/234357869-6d767c6e-45ec-4248-a009-87f31bd37f0b.png)
<img width="690" alt="image" src="https://user-images.githubusercontent.com/129091848/234358312-98247562-1646-4256-af8a-53f66a79c9fb.png">

Demo has read only datepicker field using readonly="true", this did not work for me as a solution

**Comment by tmcconechy** [comment]
@ericangeles can you see if anyone in your group might know went wrong here?

**Comment by yohannahbautista** [comment]
Hello, based on one of the datepicker example pages: http://localhost:4200/ids-enterprise-ng-demo/datepicker
Readonly looks to work best after initialization. I also found a comment on the set input which confirms it
<img width="514" alt="image" src="https://user-images.githubusercontent.com/85279127/234547470-bf4d3ef7-e5cb-4ef2-9836-80537c9113b8.png">

For dropdown though, it looks to work for readonly based on this: http://localhost:4200/ids-enterprise-ng-demo/dropdown-simple
I also checked with disabled="true" and it works as well

**Comment by pallisj** [comment]
Setting the datepicker or the dropdown as readonly = "readonly" or readonly = "true" like in the demos does not set my fields to read only. These fields are still editable with the element property saying readOnly is set to false even though in the HTML it is set to either true or read only I have tested both. The datepicker, dropdown, and checkbox elements seem to not be initializing correctly and taking in the readonly value since upgrading ids. Checkboxes I have disabled in our component.ts files and that is a usable work around.

However, I don't want to disable the datepicker or dropdowns because they appear differently as disabled fields compared to our other text element read only fields and make the page lack uniformity.

**Comment by yohannahbautista** [comment]
Hello, would it be possible to make a demo it stackblitz of it? I can see the issue with datepicker and will look into it, but the dropdown readonly and disabled looks to work fine in the demos.

**Comment by tmcconechy** [comment]
@yohannahbautista yes your right on http://localhost:4200/ids-enterprise-ng-demo/dropdown-simple it works. Maybe try and add an example where the readonly is tied to a variable in the model and see that that works.

As for http://localhost:4200/ids-enterprise-ng-demo/datepicker you can see the readonly does'nt work. I would also add one where it is set via the modal to confirm.

**Comment by pallisj** [comment]
for the dropdowns I am seeing that I can have them set to read only or not but toggling back and forth hasn't been working as expected, I will look more into this.

**Comment by tmcconechy** [comment]
@pallisj do feel free to review the PR https://github.com/infor-design/enterprise-ng/pull/1474 with any suggestions

**Comment by pallisj** [comment]
I dont believe this issue should be closed. The build advice above did not provide me with a fix and I am still having issues with the drop downs not toggling correctly. Would we be able to set up a call to get these issues resolved please?

**Comment by tmcconechy** [comment]
I don't see any evidence its not work as we expect on the PR. I suspect it might be just the way your updating your code? Will do a beta next week and you can try it with the beta build. Then we can reassess.

You might need to build both the enterprise package and the enterprise-ng package. But i cant do a beta right now

Also what are you doing differently? Maybe you can run the sample app and try and edit the example to do the "different thing"?

**Comment by pallisj** [comment]
In the demo I see on the page for dropdown-single select clicking the read only button doesn't set any dropdown to read only theyre all still able to be changed. The button just becomes readonly. Is that the expected results for that page? The multiselect seems to workas I would expect that the read only button makes the dropdown options unable to be selected. This is not the case for the single select and looks like the issue I am seeing in my project

**Comment by tmcconechy** [comment]
@yohannahbautista 

When i try: http://localhost:4200/ids-enterprise-ng-demo/dropdown and click the readony button. I would expect that example to make it readonly but doesnt seem to work? Is that the issue you mean @pallisj ?

Im trying this on main after the merge? cc @ericangeles

**Comment by pallisj** [comment]
yes this is the dropdown issue I am experiencing. It seems the single select dropdowns get a read only value when initialized but then it is unable to be toggled back and forth.
